### PR TITLE
feat: [Orchestration] Update to Orchestration Spec 0.36.1

### PR DESCRIPTION
**Pull request by MatKuhr**
## Context

AI/ai-sdk-java-backlog#119.

### Feature scope:
 
- [x] Regenerate client

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [x] ~Error handling created / updated & covered by the tests above~
- [x] Aligned changes with the JavaScript SDK
- [x] ~Documentation updated~
- [x] ~Release notes updated~

### CODE REVIEW

File: orchestration/src/main/resources/spec/orchestration.yaml [hunk 1]
```diff
@@ -455,20 +543,22 @@ components:
...
-          $ref: '#/components/schemas/AzureThreshold'
-        Violence:
-          $ref: '#/components/schemas/AzureThreshold'
+        "Hate":
+          $ref: "#/components/schemas/AzureThreshold"
+        "SelfHarm":
+          $ref: "#/components/schemas/AzureThreshold"
+        "Sexual":
+          $ref: "#/components/schemas/AzureThreshold"
+        "Violence":
```
Comment (newtork) [inline orchestration/src/main/resources/spec/orchestration.yaml:559]: (Comment)

I'm wondering what makes them using `"` for property keys here, but nowhere else.

File: orchestration/src/main/resources/spec/orchestration.yaml [hunk 2]
```diff
@@ -555,6 +658,163 @@ components:
...
+      required:
+        - type
+        - config
+      properties:
+        type:
+          type: string
+          anyOf:
+            - enum:
+                - document_grounding_service
+            - {}
```
Comment (newtork) [inline orchestration/src/main/resources/spec/orchestration.yaml:674]: (Comment/Question)
wtf is this.

**Review by newtork — COMMENTED** [review]
Waiting for response on the question, why use `0.36.1` when `0.40.1` is live on prod already.

File: orchestration/src/main/resources/spec/orchestration.yaml (hunk 2 above)
Comment (MatKuhr) [reply orchestration/src/main/resources/spec/orchestration.yaml:674]: This is done for future extensibility. Without the anyof, users may complain if the enum allows additional values in the future. Not sure how strong this argument really is though..

File: orchestration/src/main/resources/spec/orchestration.yaml (hunk 1 above)
Comment (ChristophMatthies) [reply orchestration/src/main/resources/spec/orchestration.yaml:559]: My best guess is "historical reasons". git blame tells me this is from (one of) the original commits on content filtering 8 months ago.

Comment (ChristophMatthies) [reply orchestration/src/main/resources/spec/orchestration.yaml:559]: Here's a PR that addresses this: https://github.tools.sap/AI/llm-orchestration/pull/832
(code generation seems unaffected)

File: orchestration/src/main/resources/spec/orchestration.yaml (hunk 2 above)
Comment (ChristophMatthies) [reply orchestration/src/main/resources/spec/orchestration.yaml:674]: See also this (very related) carfuffle: https://github.com/OpenAPITools/openapi-diff/issues/303

Comment (MatKuhr) [reply orchestration/src/main/resources/spec/orchestration.yaml:674]: yeah, we are also setting the flag `enumUnknownDefaultCase` for exactly this reason..

**Pull request merged**